### PR TITLE
Fix notifications

### PR DESCRIPTION
**Pull request by erikos**
https://phabricator.endlessm.com/T18310

### CODE REVIEW

**Comment by erikos** [comment]
"notificationDaemon: Fix a notification deleting itself...": looks like this has been fixed upstream already: https://git.gnome.org/browse/gnome-shell/commit/js/ui/notificationDaemon.js?id=accd24e379374b202791af2b3e065e1fca4fa406

File: js/ui/notificationDaemon.js [hunk 1]
```diff
@@ -740,6 +745,11 @@ const GtkNotificationDaemonAppSource = new Lang.Class({
         else
             this.pushNotification(notification);
 
+        if (oldNotification) {
+            oldNotification.disconnect(oldNotificationDestroyId);
+            oldNotification.destroy();
+        }
+
```
Comment (smspillaz) [inline js/ui/notificationDaemon.js]: Is there any reason why we can't disconnect and destroy the old notification earlier, before we add the new notification?

File: js/ui/notificationDaemon.js [hunk 2]
```diff
@@ -805,6 +819,7 @@ const GtkNotificationDaemon = new Lang.Class({
         source.connect('count-updated', Lang.bind(this, this._saveNotifications));
         Main.messageTray.add(source);
         this._sources[appId] = source;
+        this.emit('new-gtk-notification-source', source);
```
Comment (smspillaz) [inline js/ui/notificationDaemon.js]: This is not used

File: js/ui/notificationDaemon.js [hunk 3]
```diff
@@ -762,6 +772,10 @@ const GtkNotificationDaemonAppSource = new Lang.Class({
         }
         return [this._appId, notifications];
     },
+
+    get app() {
+        return this._app;
+    }
```
Comment (smspillaz) [inline js/ui/notificationDaemon.js]: It does not look like this is used

File: js/ui/notificationDaemon.js [hunk 4]
```diff
@@ -871,7 +886,16 @@ const GtkNotificationDaemon = new Lang.Class({
 
         invocation.return_value(null);
     },
+
+    getSourceFor: function(appId) {
+        if (this._sources[appId]) {
+            return this._sources[appId];
+        }
+        return null;
+    },
```
Comment (smspillaz) [inline js/ui/notificationDaemon.js]: It does not look like this is used

File: js/ui/notificationDaemon.js [hunk 5]
```diff
@@ -880,4 +904,13 @@ const NotificationDaemon = new Lang.Class({
...
         this._gtkNotificationDaemon = new GtkNotificationDaemon();
     },
+
+    get fdo() {
+        return this._fdoNotificationDaemon;
+    },
+
+    get gtk() {
+        return this._gtkNotificationDaemon;
+    },
```
Comment (smspillaz) [inline js/ui/notificationDaemon.js]: Now that we are using `messageTray` to check notifications, is it necessary to expose this?

**Comment by cosimoc** [comment]
Superseded by #104.

**Pull request closed without merging**